### PR TITLE
PADV-624 - Create Student component

### PR DESCRIPTION
**Pull request by sergivalero20**
## Ticket

https://agile-jira.pearson.com/browse/PADV-624

## Description

This PR adds DataTable for Students component. It also includes Pagination and Sorting attribute.

## Changes made

- [x] Add DataTable component.
- [x] Add tests.

## Screenshots

image

## How to test

- Clone the Institution Portal MFE
- Run npm install.
- Run npm run start

## Reviewers

- [x] @anfbermudezme 
- [ ] @Squirrel18

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -16,6 +16,5 @@ SEGMENT_KEY=''
 SITE_NAME=localhost
 USER_INFO_COOKIE_NAME='edx-user-info'
 APP_ID=''
-MFE_CONFIG_API_URL=''
 ACCOUNT_PROFILE_URL='http://localhost:1995/u'
-MFE_CONFIG_API_URL='http://localhost:18000/api/mfe_config/v1'
+MFE_CONFIG_API_URL='http://localhost:18000/api/mfe_config/v1'
```
Comment (anfbermudezme) [inline .env.development:20]: Blank line.

File: src/features/Students/StudentsTable/index.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import DataTable from '@edx/paragon/dist/DataTable';
+import { Row, Col } from '@edx/paragon';
+
+const StudentsTable = ({ data, columns, count }) => (
+  <Row className="justify-content-center my-4 border-gray-300 bg-light-100 my-3">
+    <Col xs={11}>
+      <DataTable isSortable itemCount={count} data={data} columns={columns}>
```
Comment (anfbermudezme) [inline src/features/Students/StudentsTable/index.jsx:9]:
```suggestion
      <DataTable 
        isSortable
        itemCount={count}
        data={data}
        columns={columns}
      >
```

File: src/features/Students/StudentsTable/index.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        <DataTable.Table />
+        <DataTable.EmptyTable content="No enrollments found." />
```
Comment (anfbermudezme) [inline src/features/Students/StudentsTable/index.jsx:11]:
```suggestion
        <DataTable.EmptyTable content="No students found." />
```

File: src/features/Students/data/slices.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    </StudentEnrollmentsContext.Provider>
+  );
+};
+
+const useStudentEnrollments = () => {
+  const context = useContext(StudentEnrollmentsContext);
+
+  if (!context) {
+    throw new Error('useStudentEnrollments must be used within a StudentEnrollmentsProvider');
+  }
```
Comment (anfbermudezme) [inline src/features/Students/data/slices.js:61]:
```suggestion
  };
```

File: src/features/Students/data/api.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { getConfig } from '@edx/frontend-platform';
+
+function getStudentbyInstitutionAdmin(page) {
+  const params = { page };
+
+  Object.keys(params).forEach((key) => {
+    if (params[key] === '') {
+      delete params[key];
+    }
```
Comment (anfbermudezme) [inline src/features/Students/data/api.js:10]:
```suggestion
    };
```

File: src/features/Students/StudentsPage/index.jsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+  const handlePagination = async (targetPage) => {
+    dispatch({ type: 'UPDATE_CURRENT_PAGE', payload: targetPage });
+
+    const fetchData = () => {
+      try {
+        fetchStudentEnrollments(dispatch, targetPage);
+      } catch (error) {
+        logError(error);
+      }
```
Comment (anfbermudezme) [inline src/features/Students/StudentsPage/index.jsx:37]:
```suggestion
      };
```

File: src/features/Students/StudentsPage/index.jsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+const StudentsPage = () => {
+  const { state, dispatch } = useStudentEnrollments();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        await fetchStudentEnrollments(dispatch);
+      } catch (error) {
+        logError(error);
+      }
```
Comment (anfbermudezme) [inline src/features/Students/StudentsPage/index.jsx:21]:
```suggestion
      };
```

**Review by anfbermudezme — CHANGES_REQUESTED** [review]

File: src/features/Students/data/slices.js (hunk 4 above)
Comment (sergivalero20) [reply src/features/Students/data/slices.js:61]: @anfbermudezme unncessery semi colon according to Lint.

File: src/features/Students/data/api.js (hunk 5 above)
Comment (sergivalero20) [reply src/features/Students/data/api.js:10]: @anfbermudezme unncessery semicolon according to Lint.

File: src/features/Students/StudentsPage/index.jsx (hunk 6 above)
Comment (sergivalero20) [reply src/features/Students/StudentsPage/index.jsx:37]: @anfbermudezme unncessery semicolon according to Lint.

File: src/features/Students/StudentsPage/index.jsx (hunk 7 above)
Comment (sergivalero20) [reply src/features/Students/StudentsPage/index.jsx:21]: @anfbermudezme unncessery semicolon according to Lint.

**Review by anfbermudezme — CHANGES_REQUESTED** [review]

File: src/features/Students/data/api.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { getConfig } from '@edx/frontend-platform';
+
+function getStudentbyInstitutionAdmin(page) {
+  const params = { page };
+
+  Object.keys(params).forEach((key) => {
```
[the remark this answers is not in the document]
Comment (sergivalero20) [reply src/features/Students/data/api.js:7]: @anfbermudezme It's not necessary. It doesn't any effect over pagination.

**Review by anfbermudezme — APPROVED** [review]

File: package.json [hunk 9]
```diff
@@ -44,6 +44,7 @@
     "prop-types": "15.8.1",
     "react": "16.14.0",
     "react-dom": "16.14.0",
+    "react-intl": "^5.25.1",
```
Comment (Squirrel18) [inline package.json:47]: Check this version in the Learning MFE. @sergivalero20

File: src/features/Students/StudentsPage/index.jsx [hunk 10]
```diff
@@ -0,0 +1,95 @@
+/* eslint-disable react-hooks/exhaustive-deps */
```
Comment (Squirrel18) [inline src/features/Students/StudentsPage/index.jsx:1]: Please, disable this message where is thrown. @sergivalero20

**Review by Squirrel18 — APPROVED** [review]

File: package.json (hunk 9 above)
Comment (sergivalero20) [reply package.json:47]: @Squirrel18 With the current version doesn't cause any conflict. I run `npm run build` and everything is fine.

**Pull request merged**